### ISSUE TITLE
Add firewall

### ISSUE THREAD

**Issue by vrsovin** [guardianproject/orbot#901]
Add a "firewall" like the netguard project, blocking the connections of all apps (user and system) that are not selected on the choose apps option.

**Comment by n8fr8** [comment]
Great idea and something we have wanted to do for a long time.

Have you tried https://rethinkdns.com/ app? It is a great firewall that integrates with Orbot for Tor proxying.

**Comment by Gamer750** [comment]
I also agree that it necessary to add firewall in orbot . Because if you use a googled phone or if you install apps like WhatsApp Facebook they collect your IP address in background and send to their server so there is no point in using orbot or VPN if they can collect IP address in background so it is necessary to add firewall in orbot

**Comment by Gamer750** [comment]
What about integrating karma firewall in orbit
https://github.com/stargw-net/karma-firewall

**Comment by shuttittuppitt** [comment]
August 14, 2025

(at)n8fr8

I have used Rethink (from here = https://play.google.com/store/apps/details?id=com.celzero.bravedns) for a _long_ time, & it _used_ to work fantastic as both a firewall & a VPN, but last week my library (which is my ONLY wifi) changed something so that Rethink's VPN is now _worthless_ (but it's firewall still works, which is why I'm still using it anyway.)

I saw in Rethink's options that it _claims_ that it can use "Orbot" (as a proxy), but that is _WRONG!_ I have both Orbot (& Rethink) installed, & running at the same time, but while Rethink's firewall is working (proof = "Software update" (which I have _blocked)_ CAN'T go online (thank God)), android's pull down notifications thingy shows that Orbot has 0 KiB/s (up & down). I can confirm that, because my internet is still censored as (guess).

I also (accidently) tested using Orbot by itself (with Rethink _accidentally_ turned "Off"), & Orbot works fantastic AS A VPN, but while a VPN is _fantastic_ for my uses, I NEED the ability to BLOCK user specified things (ESPECIALLY "Software update") from _ever_ going online.

Because Orbot _seems_ to NOT work if you're also using Rethink _at the same time_ (at least in my case), my only hope is for Orbot to get it's OWN ability to BLOCK user specified things from _ever_ going online. That way, I can uninstall Rethink & just use Orbot by itself.



(If it's important, my android tablet is _not_ rooted, because trying to root android without a PC is impossible anymore (I _have_ tried.))
